app/pulp/app/viewsets/user.py

```diff
@@ -0,0 +1,23 @@
+from pulp.app.models import PulpUser
+from pulp.app.serializers import UserSerializer
+from pulp.app.viewsets import NamedModelViewSet
+
+from rest_framework import decorators, status
+from rest_framework.response import Response
+
+class UserViewSet(NamedModelViewSet):
+    endpoint_name = 'users'
+    queryset = PulpUser.objects.all()
+    serializer_class = UserSerializer
+
+    @decorators.detail_route(methods=['post'])
+    def set_password(self, request, pk=None):
+        user = self.get_object()
+        serializer = UserSerializer(data=request.data)
+        if serializer.is_valid():
+            user.set_password(serializer.data['password'])
```

Comment:
I think this will return a 500 if the user hits the right endpoint but fails to include a password.